### PR TITLE
NRPT-457 and NRPT-459: Added migration to add BCMI flavours. Cleaned BCMI models

### PR DESCRIPTION
**Pull request by danieltruong**
https://bcmines.atlassian.net/browse/NRPT-457
https://bcmines.atlassian.net/browse/NRPT-459

### CODE REVIEW

File: api/src/models/bcmi/annualReport-bcmi.js [hunk 1]
```diff
@@ -5,11 +5,11 @@ module.exports = require('../../utils/model-schema-generator')(
   {
     _schemaName:       { type: String, default: 'AnnualReportBCMI', index: true },
     _sourceRefId:     { type: 'ObjectId', default: null, index: true },
+    _master: { type: 'ObjectId', default: null, index: true },
```
Comment (marklise) [inline api/src/models/bcmi/annualReport-bcmi.js:8]: Alignment of the braces.

File: api/src/models/bcmi/damSafetyInspection-bcmi.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ module.exports = require('../../utils/model-schema-generator')(
   {
     _schemaName:       { type: String, default: 'DamSafetyInspectionBCMI', index: true },
     _sourceRefId:     { type: 'ObjectId', default: null, index: true },
+    _master: { type: 'ObjectId', default: null, index: true },
```
Comment (marklise) [inline api/src/models/bcmi/damSafetyInspection-bcmi.js:8]: Alignment

File: api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+const CertificateAmendment = require('../src/controllers/post/certificate-amendment');
+const Correspondence = require('../src/controllers/post/correspondence');
+const DamSafetyInspection = require('../src/controllers/post/dam-safety-inspection');
+const Inspection = require('../src/controllers/post/inspection');
+const ManagementPlan = require('../src/controllers/post/management-plan');
+const Order = require('../src/controllers/post/order');
+const Permit = require('../src/controllers/post/permit');
+const Report = require('../src/controllers/post/report');
+
+var mongoose = require('mongoose');
```
Comment (KitArmstrong) [inline api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js:17]: Nit: `var`

File: api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+            break;
+          case 'Report':
+            obj = Report.createBCMI(args, null, null, flavourRecord);
+            break;
+          default:
+            break;
+        }
+
+        if (obj) {
+          promises.push(await obj.save());
```
Comment (KitArmstrong) [inline api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js:155]: I don't think the `await` should be here as it is awaiting the promise result before pushing it onto the array. They should all resolve in line 163 instead.

**Review by marklise — APPROVED** [review]

File: api/src/models/bcmi/annualReport-bcmi.js (hunk 1 above)
Comment (danieltruong) [reply api/src/models/bcmi/annualReport-bcmi.js:8]: 5d4da993f36eca4bdc1a327ab7784901d7204cc3

File: api/src/models/bcmi/damSafetyInspection-bcmi.js (hunk 2 above)
Comment (danieltruong) [reply api/src/models/bcmi/damSafetyInspection-bcmi.js:8]: 5d4da993f36eca4bdc1a327ab7784901d7204cc3

File: api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js (hunk 3 above)
Comment (danieltruong) [reply api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js]: 5d4da993f36eca4bdc1a327ab7784901d7204cc3

File: api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js (hunk 4 above)
Comment (danieltruong) [reply api/migrations/20201002173359-addBcmiFlavourToMasterRecords.js]: 5d4da993f36eca4bdc1a327ab7784901d7204cc3

**Review by KitArmstrong — APPROVED** [review]

**Pull request merged**